src/fty_common_messagebus_unix_socket_client.cc

```diff
@@ -0,0 +1,413 @@
+
+        m_pipe[0] = -1;
+        m_pipe[1] = -1;
+
+        if (pipe(m_pipe) < 0)
+        {
+            throw MessageBusException("Impossible to create the pipe: " + std::string(strerror(errno)));
+        }
+
+    }
+
+    MessageBusUnixSocketClient::~MessageBusUnixSocketClient()
+    {
+        //stop the listener
+        if(m_listenerThread.joinable())
+        {
+            write(m_pipe[1], "S", 1);
+            m_listenerThread.join();
+        }
+        
+        //close the socket
+        if(m_mainSocket != -1)
+        {
+            close(m_mainSocket);
+        }
+
+        // close the pipe
+        close(m_pipe[0]);
+        close(m_pipe[1]);
+        
+    }
+
+    void MessageBusUnixSocketClient::connect()
+    {
+        connectToServer();
+        
+        //launch the listener
+        m_listenerThread = std::thread(&MessageBusUnixSocketClient::listenerClient, this);
+    }
+
+    void MessageBusUnixSocketClient::connectToServer()
+    {
+        //setup the socket
+        m_mainSocket = socket(AF_UNIX, SOCK_STREAM, PF_UNSPEC);
+        
+        if (m_mainSocket == -1)
+        {
+            throw MessageBusException("Impossible to create unix socket "+m_socketPath+": " + std::string(strerror(errno)));
+        }
+
+        //setup the addr
+        memset(&m_addr, 0, sizeof(struct sockaddr_un));
+
+        m_addr.sun_family = AF_UNIX;
+        strncpy(m_addr.sun_path, m_socketPath.c_str(), sizeof(m_addr.sun_path) - 1);
+
+        int ret = socketConnect(m_mainSocket, (const struct sockaddr *) &m_addr, sizeof(struct sockaddr_un));
+        if (ret == -1)
+        {
+            throw MessageBusException("Impossible to connect to server using the socket "+m_socketPath+": " + std::string(strerror(errno)));
+        }
+        
+        //send the clientName
+        sendFrames(m_mainSocket, {m_clientName});
+        
+        if(recvFrames(m_mainSocket).at(0) != "OK")
+        {
+            throw MessageBusException("Impossible to connect to server using the socket "+m_socketPath+": Server refuse the connection");
+        }
+
+        log_debug("%s - connected through '%s'", m_clientName.c_str(), m_socketPath.c_str());
+    }
+
+    void MessageBusUnixSocketClient::listenerClient()
+    {
+        bool stopRequested = false;
+        
+        // Clear the reference set of socket
+        fd_set socketsSet;
+        FD_ZERO(&socketsSet);
+
+        // Add the server socket
+        FD_SET(m_mainSocket, &socketsSet);
+        int lastSocket = m_mainSocket;
+        
+        //Add the pipe
+        FD_SET(m_pipe[0], &socketsSet);
+        if (m_pipe[0] > lastSocket)
+        {
+            // Keep track of the maximum
+            lastSocket = m_pipe[0];
+        }
+
+        //infini loop for handling connection
+        for (;;)
+        {
+            fd_set tmpSockets = socketsSet;
+
+            // Detect activity on the sockets
+            if (select(lastSocket+1, &tmpSockets, NULL, NULL, NULL) == -1)
```

Comment:
This really reminds me of a `zactor_t` mainloop. Are we reinventing the wheel here?

Reply:
There's `zmq_ipc` that does inter-process communication across UNIX domain sockets, hence the lingering reinvent-the-wheel feeling.